registration form (solves: #6)

stankasvecova opened the pull request:
solves: #6

stankasvecova committed:
registration form

stankasvecova committed:
Registration.html

Autocomplete now works!

stankasvecova committed:
logo1.png

Autocomplete now works!

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+<html>
+
+<head>
+<<<<<<< Updated upstream
```
Please cleanup `<<<<<` git conflict leftovers.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+    autocomplete(document.getElementById("myInput"), countries);
+  </script>
+=======
+<title>good.teens</title>
```
why is there two forms?

BTBTravis replied:
✅

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+    </div>
+  </form>
+  <script>
+    function autocomplete(inp, arr) {
```
please move script into a js file and remember to wrap it in a function scope. 

```js
(function () {
    function autocomplete(inp, arr) {
    ...
})();
```

stankasvecova replied:
will do.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+  <title>good.teens</title>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
```
I would advise against bootstrap. Its better to learn how to quickly style things yourself. Truly mastering grid and flex will take you much further, might as well start now. 

Fully understand 
https://css-tricks.com/snippets/css/complete-guide-grid/

and 

https://css-tricks.com/snippets/css/a-guide-to-flexbox/

stankasvecova replied:
Yes, I used the grid at the online store, and now I wanted to give Bootstrap a go. Didn't know it was not advisable. I will use grid & flex from now onwards but does this bootstrap here need to be removed too?

BTBTravis replied:
For me it is not bootstrap vs grid & flex box. It is a css framework vs custom styles. I think its better to learn styling everything yourself because if you truly want to build something custom no css framework will do and it is better to write the styles yourself. 

For this page I won't block however lets move on.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
+  <style>
+    .regTitle {
```
move styles to a css file

stankasvecova replied:
Just this time, as it was just one compact thing ' registration', I thought I would have it all in one file.
Happy to split it into separate .css and .js.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+</head>
+
+<body>
+  <h1 class="regTitle"><img src="logo1.png" class=".img-rounded" style="width:200px" alt="goodteens"></h1>
```
don't inline styles. use css file.

stankasvecova replied:
OK

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+      </div>
+      <div class="col-sm-3">
+        <label for="validationCustom03">Your age in years</label>
+        <input type="text" size=65 name="yearuser" class="form-control" id="validationCustom03" placeholder="15"
```
I think type number will suit you here

stankasvecova replied:
will amend.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+        <input type="text" size=65 name="yearuser" class="form-control" id="validationCustom03" placeholder="15"
+          required>
+      </div>
+      <div class="col-sm-3">
```
this bloat html is another reason to avoid bootstrap

stankasvecova replied:
OK

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+      </div>
+      <div class="col-sm-3">
+        <label for="validationCustom06">Email address:</label>
+        <input type="text" size=65 name="email" class="form-control" id="validationCustom06" value="" required>
```
type email will suit you here

stankasvecova replied:
will amend.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+      </div>
+      <div class="col-sm-3">
+        <label for="validationCustom07">Confirm email address:</label>
+        <input type="text" size=65 name="confirmemail" class="form-control" id="validationCustom07" value="" required>
```
also type email here

stankasvecova replied:
will amend.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+      </div>
+      <div class="col-sm-3">
+        <label for="validationCustom01">Password:</label>
+        <input type="text" size=65 name="passworduser" class="form-control" id="validationCustom08" value="" required>
```
type password would be better

stankasvecova replied:
will amend.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+      var currentFocus;
+
+      inp.addEventListener("input", function (e) {
+        var a, b, i, val = this.value;
```
there are bad variable names. It should be clear what something is by it's name. what is a or b? Only exception is `i` inside a for loop

stankasvecova replied:
Well, I tried to understand that code and went thru it a few times. 
There is the concept of .this (and other stuff) that seemed too difficult and so I tried whether it would work anyway, and it did. I planned to go back to it, and analyse it further to understand it. Happy to improve it, once I do.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,382 @@
+
+    var countries = ["Afghanistan", "Albania", "Algeria", "American Samoa", "Andorra", "Angola", "Anguilla", "Antarctica", "Antigua and Barbuda", "Argentina", "Armenia", "Aruba", "Australia", "Austria", "Azerbaijan", "Bahamas", "Bahrain", "Bangladesh", "Barbados", "Belarus", "Belgium", "Belize", "Benin", "Bermuda", "Bhutan", "Bolivia", "Bosnia and Herzegowina", "Botswana", "Bouvet Island", "Brazi ...
+
+    autocomplete(document.getElementById("myInput"), countries);
```
very strong feeling you copied and pasted this code without understanding it...

stankasvecova replied:
See above.

BTBTravis reviewed (changes requested):
Nice start on the registration page @stankasvecova ! Left some comments of things that can be improved before we move on.

stankasvecova committed:
registration updated

stankasvecova committed:
Update registration.html

BTBTravis reviewed (changes requested):
The input types were changed in the wrong manner

```html
<input type="number" ...
<input type="email" ...
```
more here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,104 @@
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
```
Don't seen need for jQuery on the site or bootstrap on the site.

BTBTravis replied:
Was reading https://www.swyx.io/writing/js-third-age/ this morning. Made me think back to the jQuery days. I would focus on new tech as you continue to expand your JavaScript skills.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,104 @@
+            <input type="password" size=65 name="passworduser" class="form-control" id="validationCustom08" placeholder="type password" required>
+            <div class="form-check">
+                <input name="TAC" class="form-check-input" type="checkbox" id="gridCheck">
+                <label class="form-check-label" for="gridCheck">I agree with Terms & Conditions</label>
```
html indentation is messed up here

MrTim replied:
if you really want to have this Agree check => you need to provide those T&C

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,104 @@
+            </div>
+            <div class="col-sm-3">
+            <label for="validationCustom06">Email address:</label>
+            <input type="text" size=65 name="email" class="form-control" id="validationCustom06" placeholder="type email address" required>
```
```html
<input type="email" ...
``` 
will give us some decent native validation of the form feild

stankasvecova committed:
Update registration.html

stankasvecova committed:
Update registration.html

stankasvecova committed:
Merge branch 'registration' of https://github.com/redi-js-teachers/js_sprint_2020_final_project_stankasvecova into registration

MrTim commented on registration.html:
```diff
@@ -0,0 +1,104 @@
+            </div>
+            <div class="col-sm-3">
+            <label for="validationCustom03">Your age in years</label>
+            <input type="text" size=65 name="yearuser" class="form-control" id="validationCustom03" placeholder="type number" required>
```
just age

MrTim commented on registration.html:
```diff
@@ -0,0 +1,104 @@
+            <div class="col-sm-3">
+            </div>
+            <div class="col-sm-3">
+            <label for="validationCustom04">City</label>
```
where these strange id's are coming from?

stankasvecova committed:
changed ids, input types and T&Cs clickable

stankasvecova reviewed (commented):
**My last push should cover all Ivan's comments:**
1. why strange IDs: as I believed they would be used in back-end for some validation, but true, don't need them this complicated. 
2. T&C will be added once I have it. For now an empty <a href=""></a>

I did not change bootstrap html into grid yet, i wasn't aware of it being not a good choice. I would continue with the rest of the pages with grid though. Let me know if it needs to be changed also in the registration.

BTBTravis commented on registration.html:
```diff
@@ -0,0 +1,107 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+    <link rel="stylesheet" href="registration.css">
+    <script src="reg2.js"></script>
```
this is not the place to attach scripts and I don't think this script exist but we can fix that in future prs

stankasvecova replied:
Does exist now and script attached correctly now. Thanks

BTBTravis reviewed (dismissed):
Looking good!

MrTim reviewed (dismissed):
Looks good! Please merge this one.

stankasvecova committed:
Added autocomplete js

stankasvecova reviewed (commented):
Before I merge this one, I wanted to add the autocomplete javascript first. 
I understand the code (except a part of it which I'm attaching at the bottom of this message). The parts I did not understand, I experimented by leaving it out and it still works. I renamed variables into something more meaningful. I made the code shorter and simpler without compromising the functionality - it does work. Are you happy if it stays like this?
---
the parts of the code I didn't get:
inp.addEventListener("keydown", function(e) {  <!-- This is all clear, understoon -->
  var x = document.getElementById(this.id + "autocomplete-list"); <!-- This is all clear, understoon -->
  if (x) x = x.getElementsByTagName("div");<!-- Starting here, not clear, I don't get it -->
  if (e.keyCode == 40) {
    currentFocus++;
    addActive(x);
  } else if (e.keyCode == 38) {
    currentFocus--;
    addActive(x);
  } else if (e.keyCode == 13) {
    e.preventDefault();
    if (currentFocus > -1) {
      if (x) x[currentFocus].click();
    }
  }
function addActive(x) {
if (!x) return false;
removeActive(x);
if (currentFocus >= x.length) currentFocus = 0;
if (currentFocus < 0) currentFocus = (x.length - 1);
x[currentFocus].classList.add("autocomplete-active");
}
function removeActive(x) {
for (var i = 0; i < x.length; i++) {
  x[i].classList.remove("autocomplete-active");
}
}

MrTim commented on registration.css:
```diff
@@ -0,0 +1,29 @@
+	margin: 1em;
+	border-radius: 1em;
+}
+.inp {
```
looks like your not using this style. If it's true => remove it

stankasvecova replied:
Removed

MrTim commented on registration.html:
```diff
@@ -0,0 +1,107 @@
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+    <link rel="stylesheet" href="registration.css">
```
you can put css file under css folder which is already in your project

stankasvecova replied:
Will do.

MrTim commented on registration.html:
```diff
@@ -0,0 +1,107 @@
+            </div>
+        </div>
+    </form>
+    <script src="registration.js"></script>
```
the same with js files - please put all js files under js folder, so you would have more solid structure for the project

stankasvecova replied:
will do.

MrTim commented on registration.html:
```diff
@@ -0,0 +1,107 @@
+                </div>
+                <p></p>
+                <div class="col-sm-3">
+                    <button class="btn btn-danger" type="submit">Create an account</button></div>
```
where is btn-danger style is coming from? I can't find it...

stankasvecova replied:
From bootstrap pre-set styles in the <link ref=""> on line 8. But it's true, i don't need to have it doubled, btn style is enough, so btn-danger style is now removed.

MrTim commented on registration.html:
```diff
@@ -0,0 +1,107 @@
+<body>
+    <h1 class="regTitle"><img src="logo1.png" class="gtlogo" alt="goodteens"></h1>
+    <h1 class="regTitle"> Register a new account </h1>
+    <form name="RegForm" action="process.php" method="post" autocomplete="off">
```
which action you want to perform on submitting the form?

stankasvecova replied:
Posting the inputed data into the process.php file.

MrTim commented on registration.js:
```diff
@@ -0,0 +1,51 @@
+function autocomplete(inp, arr) {
+
+var currentFocus;
```
two things:

1. you have problem with indentation. Please move all function content right
2. why it's `var`?

stankasvecova replied:
ran auto-format function on it so hopefully fixed. 
'var's - didn't poke my eye, my bad. Changed into 'let's.

MrTim commented on registration.js:
```diff
@@ -0,0 +1,51 @@
+  currentFocus = -1;
+  
+  countryDiv = document.createElement("DIV");
+  countryDiv.setAttribute("id", this.id + "autocomplete-list");
```
do you understand what `this` keyword is stands for in here?

stankasvecova replied:
yes, we are setting the "id" for the countryDiv, and this.id relates to the current created countryDiv?

MrTim commented:
> ## Before I merge this one, I wanted to add the autocomplete javascript first.
> I understand the code (except a part of it which I'm attaching at the bottom of this message). The parts I did not understand, I experimented by leaving it out and it still works. I renamed variables into something more meaningful. I made the code shorter and simpler without compromising the functionality - it does work. Are you happy if it stays like this?
> 
> the parts of the code I didn't get:
> inp.addEventListener("keydown", function(e) {
> var x = document.getElementById(this.id + "autocomplete-list");
> if (x) x = x.getElementsByTagName("div");
> if (e.keyCode == 40) {
> currentFocus++;
> addActive(x);
> } else if (e.keyCode == 38) {
> currentFocus--;
> addActive(x);
> } else if (e.keyCode == 13) {
> e.preventDefault();
> if (currentFocus > -1) {
> if (x) x[currentFocus].click();
> }
> }
> function addActive(x) {
> if (!x) return false;
> removeActive(x);
> if (currentFocus >= x.length) currentFocus = 0;
> if (currentFocus < 0) currentFocus = (x.length - 1);
> x[currentFocus].classList.add("autocomplete-active");
> }
> function removeActive(x) {
> for (var i = 0; i < x.length; i++) {
> x[i].classList.remove("autocomplete-active");
> }
> }

regarding this:
first - you can better attach code samples in github comments. Please take a look at this link https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks

About question itself
`keyCode == 38` is an arrow key `up` and `keyCode == 40` is an arrow key `down` which gives a user possibility to choose needed country with up down keys without a mouse. 
If you want to use this code - it's fine. But please fix all variable keywords. It would be nice to not move the content of the main page, but I think for now we can leave it as it is.

stankasvecova committed:
moved to correct folders

stankasvecova committed:
deleted to be moved to respective folders

stankasvecova committed:
Work in progress: design

A complete start, I don't like it yet. The two dividers will be tilted/skewed/curved when i figure out how to do it. Might get a different grey as background.

BTBTravis commented on docs/hobbies.html:
```diff
@@ -0,0 +1,113 @@
+<!DOCTYPE html>
```
First this is in the wrong folder. `/docs` is for documentation markdown files and images for those files. Also this pull request is only about registration form please remove this file from the pull request. You may submit it in another

MrTim replied:
@stankasvecova please move it into html folder. 
And this part has nothing to do with registration, so please remove it from here and continue with it in different branch.

stankasvecova replied:
Done

BTBTravis commented on js/registration.js:
```diff
@@ -0,0 +1,51 @@
+function autocomplete(inp, arr) {
```
I think it is a mistake to include this javascript. #6 asks for a static html css registration form. Lets keep things simple and expand on them with javascript in future PRs

MrTim replied:
It's too late to remove the js from here. I did review already and need to do a final review and it would be merged.

BTBTravis reviewed (changes requested):
hey @stankasvecova lets reduce the size of this pr so we can get it merged and move on

MrTim commented on docs/registration.html:
```diff
@@ -0,0 +1,107 @@
+<!DOCTYPE html>
```
Please move this one into `html` forlder.

stankasvecova replied:
done

MrTim reviewed (changes requested):
Please remove hobbies from this pr and put registration.html in html folder and after that this PR will be ready to merge.

stankasvecova committed:
Delete registration.html

stankasvecova committed:
Delete hobbies.html

stankasvecova committed:
Delete homepage_proposal.png

stankasvecova committed:
Readme file

stankasvecova committed:
Delete hobbies.css

stankasvecova committed:
final updates registration

MrTim reviewed (approved):
Nice, now it's better!
Only thing you need to fix is the css and logo imports
`<link rel="stylesheet" href="registration.css">` because now they are in different folders.
But you can do it and after merge!

BTBTravis reviewed (approved):
👍
